mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -1428,6 +1477,16 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 	return resourceSqlDatabaseInstanceRead(d, meta)
 }
 
+func maintenanceVersionDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
+	// Ignore the database version part and only compare the last part of the maintenance version which represents the release date of the version.
+	if len(old) > 14 && len(new) > 14 && old[len(old)-14:] > new[len(new)-14:] {
```

Comment:
Question: Do you think if we should suppress when `old[len(old)-14:] = new[len(new)-14:]`?

Reply:
Good question. If the version is same, it's a no-op on the backend. Having said that, I think we can also suppress on Terraform side when it's equal. I have added that logic.